index.js

```diff
@@ -1,30 +1,194 @@
-main(input, 30);
+
+const generateFromQueue = async () => {
+    let data;
+    try {
+        data = await getMessage();
+    } catch (err) {
+        console.error('Unable to get message from queue:', err);
+        setTimeout(generateFromQueue, 10000);
+        return;
+    }
+
+    if (!data.Messages) {
+        console.error('No messages in queue');
+        generateFromQueue();
+        return;
+    }
+
+    let message = data.Messages[0];
+    let messageBody;
+    try {
+        messageBody = JSON.parse(message.Body);
+        if (!isValidMessage(messageBody)) {
+            throw new Error('Invalid message from queue');
+        }
+    } catch (err) {
+        console.error(err);
+        sendToDLQ(message);
```

Comment:
i am not expert, but I thought DLQ should be automatically appended if handler return error code

Reply:
The way I configured the queue itself was that after a message has been pulled from the queue and attempted to be processed 3 times it would be sent to the DLQ. 

This would allow for some errors in processing that might be non-deterministic such as the API returning an invalid flashcard format or a set of flashcards that are less than are required number since it would be ok for processing to be attempted again.

However in this case I wrote `sendToDLQ` to handle errors that were deterministic, in this case the message being malformed. I did this because putting it back on the queue would always result in it failing 2 more times anyway. 

There may be a better way to do this such as modifying the queue to only allow messages to be pulled 1 time and add the retry logic in the `generate` function itself or just to not worry about sending to DLQ directly since it would end up there after 2 more attempts anyway.

Reply:
gotcha